src/concepts/workspan.md

```diff
@@ -1 +1,497 @@
+
+<center> \(W_{length}(n) = (\sum_{i=1}^n c_0) + c_1\) </center>
+
+We will usually omit the \(c_1\), since it does not matter asymptotically. Then, clearly this is equivalent to \(nc_0 + c_1\). We see that this closed-form solution is linear in \(n\) - so then we have that the work and span of this function is in \(O(n)\), which is consistent with what we would expect if we had "eyeballed" it.
+
+## Work/Span Analysis: Trees
+First, we will discuss the definition of a binary tree in SML:
+```sml
+datatype tree = Empty 
+              | Node of tree * int * tree
+```
+
+This denotes that a tree is either the constant constructor `Empty` denoting the empty tree, or a `Node` that contains an integer value, as well as two `tree` children, that can themselves be `Node`s or `Empty`.
+
+<figure class="aligncenter">
+    <img src="assets/treeTrans.png" alt="Tree" width="500"/>
+    <figcaption><b>Fig 2.</b> Sample binary tree</figcaption>
+</figure>
+
+So for instance, we may represent the above tree with `Node(Node(Node(Empty, 4, Empty), 3, Empty), 1, Node(Empty, 2, Empty))`. Put more fancily:
+```sml
+Node(
+    Node(
+        Node(
+            Empty,
+            4,
+            Empty
+        ),
+        3,
+        Empty
+    ),
+    1,
+    Node(
+        Empty,
+        2,
+        Empty
+    )
+)
+```
+
+Now we will analyze the complexity of finding the size of a tree. Consider the
+following implementation for doing so:
+```sml
+fun size (Empty : tree) : int = 0
+  | size (Node(L, x, R) : tree) : int = size L + 1 + size R
+```
+
+First convince yourself that it actually works. It simply recursively finds the
+size of the left and right tree, then adds one for the node that it is currently
+at. In the empty case, we consider the empty tree to have a size of 0.
+
+The major difference between this function and the previous `length` function
+was that `length` had one recursive call - `size` has two. We will need to
+reflect this change when we write our recurrences. Additionally, we need a new
+variable for our recurrence - we no longer have a list whose length we can
+induct on. A similar analogue will be \(n\), the number of nodes in the tree, so
+we will take that as our recurrence variable. We will focus first on work.
+
+We will obtain the following work recurrence:
+
+<center> \(W_{size}(n) = c_0 + W_{size}(n_l) + W_{size}(n_r)\) </center>
+<center> \(W_{size}(0) = c_1\) </center>
+
+where we define the number of nodes in the tree \(n = 1 + n_l + n_r\), and
+\(n_l\) and \(n_r\) denote the number of nodes in the left and right subtree,
+respectively. This follows similarly to our recurrence for `length` in the
+previous part, where `c_0` is just some constant amount of work that we
+necessarily have to do, and the two \(W_{size}\) calls are from the two
+recursive calls we make to `L` and `R`. 
+
+Now, we don't know precisely how big \(n_1\) and \(n_r\) are, with respect to
```

Comment:
```suggestion
Now, we don't know precisely how big \(n_l\) and \(n_r\) are, with respect to
```
Is this what you meant?

Reply:
did i type an n_1 when i meant n_l? i dont actually even comprehend how that's possible, but NICE catch